### PR TITLE
Update dockerfiles and compose files to use master branches and ROS

### PR DESCRIPTION
**Pull request by mfussi66**
This PR aims to update the dockerfiles in order to build images with:
- latest master of superbuild, and latest master of yarp
- Addition of ROS in order to slowly transition towards usage of TF

Regarding speech interaction containers:
- The docker image used for speech interaction starts off from ubuntu 20.04 instead of 22.04 to keep compatibility with `grpc 1.3`.
- The `googleSpeech` and `googleSpeechProcess` applications are now launched from yarpmanager, inside a container that launches only  `yarp run`: this is done to avoid unrealiable behaviour of fast_tcp carrier

The building process was successfully tested both with the github action and usage on the field.

### CODE REVIEW

File: docker/compose/docker-compose-console.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 version: "3.7"
 
 x-fdg-basic: &fdg-basic
-  image: ghcr.io/robotology/assistive-rehab:v2022.05.2-basic
+  image: ghcr.io/robotology/assistive-rehab:master-basic-devel
```
Comment (pattacini) [inline docker/compose/docker-compose-console.yml:4]: We don't yield any `...-basic-devel` image via the action.

File: docker/compose/docker-compose-replay.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ x-yarp-openpose: &yarp-openpose
 
 # Create alias for container running assistive-rehab modules
 x-fdg-basic: &fdg-basic
-  image: ghcr.io/robotology/assistive-rehab:v2022.05.2-basic
+  image: ghcr.io/robotology/assistive-rehab:master-basic-devel
```
Comment (pattacini) [inline docker/compose/docker-compose-replay.yml:26]: Same as above.

File: docker/images/basic/Dockerfile [hunk 3]
```diff
@@ -1,15 +1,17 @@
-ARG SOURCE_IMG="ubuntu:focal"
+# docker build -t ghcr.io/robotology/assistive-rehab:master-basic .
+
+ARG SOURCE_IMG="ubuntu:jammy"
 FROM $SOURCE_IMG as builder
 
-LABEL maintainer="valentina.vasco@iit.it, alexandre.gomespereira@iit.it, vadim.tikhanoff@iit.it" 
+LABEL maintainer="mattia.fussi@iit.it, ettore.landini@iit.it, marco.randazzo@iit.it steano.bernagozzi@iit.it"
```
Comment (pattacini) [inline docker/images/basic/Dockerfile:6]:
```suggestion
LABEL maintainer="mattia.fussi@iit.it, ettore.landini@iit.it, marco.randazzo@iit.it, steano.bernagozzi@iit.it"
```

Missing comma.

File: docker/images/navigation/Dockerfile [hunk 4]
```diff
@@ -1,7 +1,9 @@
-ARG SUPERBUILD_RELEASE="v2022.05.2"
+# docker build -t ghcr.io/robotology/assistive-rehab:master-navigation .
+
+ARG SUPERBUILD_RELEASE="master"
 FROM ghcr.io/robotology/assistive-rehab:${SUPERBUILD_RELEASE}-basic as builder
 
-LABEL maintainer="valentina.vasco@iit.it, alexandre.gomespereira@iit.it, vadim.tikhanoff@iit.it" 
+LABEL maintainer="mattia.fussi@iit.it, ettore.landini@iit.it, marco.randazzo@iit.it steano.bernagozzi@iit.it"
```
Comment (pattacini) [inline docker/images/navigation/Dockerfile:6]:
```suggestion
LABEL maintainer="mattia.fussi@iit.it, ettore.landini@iit.it, marco.randazzo@iit.it, steano.bernagozzi@iit.it" 
```

Missing comma.

**Review by pattacini — CHANGES_REQUESTED** [review]
I think there's a problem with the name of the basic image.
We don't apply any suffix to it.

File: docker/compose/docker-compose-console.yml (hunk 1 above)
Comment (mfussi66) [reply docker/compose/docker-compose-console.yml]: Fixed by removing the `-devel` suffix, good point.

File: docker/compose/docker-compose-replay.yml (hunk 2 above)
Comment (mfussi66) [reply docker/compose/docker-compose-replay.yml]: Fixed by removing the `-devel` suffix

**Review by pattacini — APPROVED** [review]

**Pull request merged**